app/models/reporting/decisions.rb

```diff
@@ -5,15 +5,132 @@
 ###
 
 class Reporting::Decisions < ApplicationRecord
+  include ArelHelper
+
   belongs_to :client, foreign_key: :cas_client_id
 
-  scope :started_between, -> (start_date:, end_date:) do
+  scope :started_between, ->(start_date:, end_date:) do
     where(match_started_at: (start_date..end_date))
   end
 
-  scope :ended_between, -> (start_date:, end_date:) do
+  scope :ended_between, ->(start_date:, end_date:) do
     terminated.
-    where(updated_at: (start_date..end_date + 1.day))
+      where(updated_at: (start_date..end_date + 1.day))
+  end
+
+  scope :open_between, ->(start_date:, end_date:) do
+    at = arel_table
+    # Excellent discussion of why this works:
+    # http://stackoverflow.com/questions/325933/determine-whether-two-date-ranges-overlap
+    d_1_start = start_date
+    d_1_end = end_date
+    d_2_start = at[:match_started_at]
+    d_2_end = at[:updated_at]
+    # Currently does not count as an overlap if one starts on the end of the other
+    where(d_2_end.gteq(d_1_start).or(d_2_end.eq(nil)).and(d_2_start.lteq(d_1_end)))
+  end
+
+  scope :on_route, ->(limit) do
+    route_names = Array.wrap(limit).map do |route_id|
+      MatchRoutes::Base.find(route_id).title
+    end
+    where(match_route: route_names)
+  end
+
+  scope :program_type, ->(limit) do
+    where(program_type: limit)
+  end
+
+  scope :associated_with_agency, ->(limit) do
+    # joins are hard through polymorphic associations...
+    program_names = EntityViewPermission.where(entity_type: 'Program', editable: true, agency_id: limit).
```

Comment:
can you `joins(:entity)` or `preload(:entity)`?

Reply:
A join results in `ActiveRecord::EagerLoadPolymorphicError: Cannot eagerly load the polymorphic association :entity` and preload doesn't actually do the load... This is where the manual join processing in the report framework comes from -- we can do it here, as we know they will all be programs, but, that seemed like an optimization pass to me.